### PR TITLE
BUG: imsave searches for wrong channel axis if image has 3 or 4 rows or columns

### PR DESCRIPTION
**Pull request by chemelnucfin**
Fixed bug in imsave if there are 3 or 4 rows/columns - now imsave requires the channel axis to be the 3rd value as specified in docs.

In toimage, channel_axis was not specified, and if (3 in shape) or if 4 was in shape, the first 3 or 4 will be taken as the channel_axis.  Which is wrong if it was a MxNx3 matrix or MxNx4 matrix if either M or N ==   3 or 4.

Added test and some sample pictures.

### CODE REVIEW

File: scipy/misc/tests/test_pilutil.py [hunk 1]
```diff
@@ -65,30 +66,31 @@ def test_bytescale_keywords(self):
...
-
-
+                img = misc.imread(png)
+            tmpdir = tempfile.mkdtemp()
+            try:
+                fn1 = os.path.join(tmpdir, 'test.png')
+                fn2 = os.path.join(tmpdir, 'testimg')
+                with warnings.catch_warnings(record=True):  # PIL ResourceWarning
+                    misc.imsave(fn1, img)
+                    misc.imsvae(fn2, img, 'PNG')
```
Comment (rgommers) [inline scipy/misc/tests/test_pilutil.py]: typo ``imsvae``

**Comment by rgommers** [comment]
Thanks @chemelnucfin. Looks good to me except for the typo.

**Comment by chemelnucfin** [comment]
Oops.  You are right, I saw that from the build.  I corrected it and
submitted again.

I am not able to runtests.py locally, and made a mistake removing the
"imwrite" portion of the test and using local packages.  My fault.

Alan

**Comment by rgommers** [comment]
All green now, so in it goes. Thanks @chemelnucfin

**Pull request merged**